api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -128,19 +125,9 @@ public void setUserProvider(Provider<User> userProvider) {
   @Override
   public ResponseEntity<ClusterLocalizeResponse> localize(
       String projectName, String clusterName, ClusterLocalizeRequest body) {
-    org.pmiops.workbench.firecloud.model.Workspace fcWorkspace;
-    try {
-      fcWorkspace = fireCloudService.getWorkspace(body.getWorkspaceNamespace(), body.getWorkspaceId())
-          .getWorkspace();
-    } catch (ApiException e) {
-      if (e.getCode() == 404) {
-        log.log(Level.INFO, "Firecloud workspace not found", e);
-        throw new NotFoundException(String.format(
```

Comment:
I think it would still be worth catching `NotFoundException` and rethrowing with this improved message.